### PR TITLE
Fix range field default value

### PR DESCRIPTION
**Pull request by afbora**
## Describe the PR

@distantnative Need your review for this simple PR :))

**Fixed Preview Field**

```
fields:
  perPage:
    label: Einträge pro Seite
    type: range
    step: 3
    min: 3
    max: 30
    default: 12
    required: true
    translate: false
```

FireShot Capture 746 - Test - Mægazine - http___localhost_test_kirby_1445_panel_pages_notes+test

## Related issues
- Fixes https://github.com/getkirby/kirby/issues/1445

## Todos

- [ ] Add unit tests for fixed bug/feature
- [ ] Pass all unit tests
- [ ] Fix code style issues with CS fixer and `composer fix`
- [ ] If needed, in-code documentation (DocBlocks etc.)

### CODE REVIEW

File: panel/src/components/Forms/Input/RangeInput.vue [hunk 1]
```diff
@@ -35,6 +35,7 @@ export default {
     autofocus: Boolean,
     disabled: Boolean,
     id: [String, Number],
+    default: [Number, String],
```
Comment (afbora) [inline panel/src/components/Forms/Input/RangeInput.vue:38]: `this.default` was returning `undefined` without this prop.

File: panel/src/components/Forms/Input/RangeInput.vue [hunk 2]
```diff
@@ -70,14 +71,14 @@ export default {
   },
   computed: {
     label() {
-      return this.value !== null ? this.format(this.value) : "–";
+      return this.required || this.value ? this.format(this.position) : "–";
```
Comment (afbora) [inline panel/src/components/Forms/Input/RangeInput.vue:74]: Only if required and value is not empty, the position is written to the label.

File: panel/src/components/Forms/Input/RangeInput.vue [hunk 3]
```diff
@@ -70,14 +71,14 @@ export default {
...
     },
     center() {
       const middle = (this.max - this.min) / 2 + this.min;
       return Math.ceil(middle / this.step) * this.step;
     },
     position() {
-      return this.value !== null ? this.value : this.center;
+      return this.value || this.default || this.center;
```
Comment (afbora) [inline panel/src/components/Forms/Input/RangeInput.vue:81]: It always shows the correct position, whether it is required field or not.

**Comment by distantnative** [comment]
It looks good to me but I haven't been able to test it myself.

**Comment by bastianallgeier** [comment]
I made a few small changes. The computed position value should be used in all places. Then it works as expected.

**Comment by plan44** [comment]
Sorry for posting here but it seems closely related:
For a range with min=0 and max=5, when the stored value is 0, it does not display the value zero (neither on the slider nor in the label) but the default value.

It seems to me that this is a JS zero/null confusion caused by the code changes above.

The original line 80 in RangeInput.vue explicitly compared with null, where the change now only uses `this.value` which is falseish also when it contains the number zero.

I tried to test/fix it in my kirby installation but apparently the vue sources are not directly used but need some build step I have no idea how to perform.

**Comment by afbora** [comment]
Thank you @plan44, i have fixed the issue on #2549

**Pull request merged**